### PR TITLE
Rq migration

### PR DESCRIPTION
**Pull request by arifBurakDemiray**
Please have a look at to the MigrationHelperTests class to have an idea about what this migration must achieve

Tests PR - https://github.com/Countly/countly-sdk-android/pull/306

### CODE REVIEW

File: sdk/src/main/java/ly/count/android/sdk/Utils.java [hunk 1]
```diff
@@ -370,4 +372,54 @@ static String[] extractValueFromString(@NonNull String data, @NonNull String sta
         // if startStr does not exist just return empty string[]
         return new String[] { data, null };
     }
+
```
Comment (arifBurakDemiray) [inline sdk/src/main/java/ly/count/android/sdk/Utils.java:375]: create a new PR for this and also add test cases

Comment (arifBurakDemiray) [reply sdk/src/main/java/ly/count/android/sdk/Utils.java:375]: https://github.com/Countly/countly-sdk-android/pull/303

**Review by peterBrxwn — APPROVED** [review]

**Review by turtledreams — CHANGES_REQUESTED** [review]
le conflicts

File: sdk/src/androidTest/java/ly/count/android/sdk/ModuleRemoteConfigTests.java [hunk 2]
```diff
@@ -88,12 +88,14 @@ public void automaticRCTriggers() {
...
+            Assert.assertEquals(++intendedCount, triggerCounter[0]);//changing device ID with merging should create a request
 
             countly.deviceId().changeWithoutMerge("dd11");
-            //todo the current behaviour is slightly out of spec as it would download RC after the RQ would have executed that request
-            //todo this should be updated once the RQ is reworked
-            Assert.assertEquals(intendedCount, triggerCounter[0]);//changing device ID without merging should create a request
+            if (a == 0) {
+                Assert.assertEquals(intendedCount, triggerCounter[0]);//changing device ID without merging won't create a request because all consents are revoked
+            } else {
+                Assert.assertEquals(++intendedCount, triggerCounter[0]);//changing device ID without merging would create a request because consent is not required
```
Comment (turtledreams) [inline sdk/src/androidTest/java/ly/count/android/sdk/ModuleRemoteConfigTests.java:97]: with?

File: sdk/src/androidTest/java/ly/count/android/sdk/TemporaryIDTests.java [hunk 3]
```diff
@@ -29,12 +29,11 @@ public void tearDown() {
         store.clear();
     }
 
-    //String temporaryIdOverrideTag = "&override_id=" + DeviceId.temporaryCountlyDeviceId;
     //String temporaryIdTag = "&device_id=" + DeviceId.temporaryCountlyDeviceId;
 
     @NonNull String[] CreateInitialTmpIDState(@NonNull String did, @NonNull DeviceIdType dType, @NonNull String replaceDid, @NonNull CountlyStore cStore) {
         String req1 = "aa=45&device_id=" + DeviceId.temporaryCountlyDeviceId;
-        String req2 = "12=qw&device_id=55";
+        String req2 = "12=qw&device_id=55"; // this is a merge request and because of migration 9 that parameter is added
```
Comment (turtledreams) [inline sdk/src/androidTest/java/ly/count/android/sdk/TemporaryIDTests.java:36]: 9? 😱

**Review by turtledreams — CHANGES_REQUESTED** [review]

File: sdk/src/androidTest/java/ly/count/android/sdk/TemporaryIDTests.java (hunk 3 above)
Comment (arifBurakDemiray) [reply sdk/src/androidTest/java/ly/count/android/sdk/TemporaryIDTests.java:36]: 😅

File: sdk/src/androidTest/java/ly/count/android/sdk/ModuleRemoteConfigTests.java (hunk 2 above)
Comment (arifBurakDemiray) [reply sdk/src/androidTest/java/ly/count/android/sdk/ModuleRemoteConfigTests.java:97]: they are both without, it is just that that test case is written in a for loop to test multiple scenarios. So I did not break that behavious

**Review by turtledreams — APPROVED** [review]

**Pull request merged**